src/client/Client.h

```diff
@@ -1299,4 +1335,98 @@ class StandaloneClient : public Client
   void shutdown() override;
 };
 
+class destructive_lock_ref_t {
```

Comment:
This needs some comments explaining what this thing is, and the rationale for it.

Reply:
Sure, I will add more comments to explain the code.